tools/Formaline.sh

```diff
@@ -8,8 +8,18 @@
 formaline_archive_name=spectre_$1
 formaline_dir=@CMAKE_BINARY_DIR@/tmp/
 pushd @CMAKE_SOURCE_DIR@ >/dev/null
-git ls-tree --full-tree --name-only HEAD \
-    | xargs tar -czf ${formaline_dir}/${formaline_archive_name}.tar.gz
+if [ -d "./.git" ]; then
+    git ls-tree --full-tree --name-only HEAD \
+        | xargs tar -czf ${formaline_dir}/${formaline_archive_name}.tar.gz
+else
+    # Since we can't record all the files in the repo (because there is
+    # no repo), we have to manually save a list of files to avoid trying
+    # to archive build dirs created inside the source dir.
+    tar -czf ${formaline_dir}/${formaline_archive_name}.tar.gz \
+        .clang-format cmake CMakeLists.txt containers docs external \
+        .github .gitignore LICENSE.txt README.md src .style.yapf \
+        support tests tools .travis .travis.yml
```

Comment:
oof... don't love having to keep this updated by hand, but I agree we need something like this to enable reformalining after deformalining... 🤷

Reply:
I was thinking about that and we could maintain this list in CMake and then have CI check that compares the list to what git says the list should be